### ISSUE TITLE
`Distribution.requires` appears incomplete in some (?) cases

### ISSUE THREAD

**Issue by jaraco** [python/importlib_metadata#67]
In GitLab by [@asottile](https://gitlab.com/asottile) on Jun 4, 2019, 21:35

```pycon
>>> import importlib_metadata
>>> for r in importlib_metadata.distribution('celery').requires:
...     print(r)
... 
pytz (>dev)
>>> for r in importlib_metadata.distribution('celery').metadata.get_all('requires-dist'):
...     print(r)
... 
pytz (>dev)
billiard (<4.0,>=3.6.0)
kombu (<5.0,>=4.4.0)
vine (>=1.3.0)
pyArango (>=1.3.2) ; extra == 'arangodb'
cryptography ; extra == 'auth'
azure-storage (==0.36.0) ; extra == 'azureblockblob'
azure-common (==1.1.5) ; extra == 'azureblockblob'
azure-storage-common (==1.1.0) ; extra == 'azureblockblob'
brotli (>=1.0.0) ; (platform_python_implementation == "CPython") and extra == 'brotli'
brotlipy (>=0.7.0) ; (platform_python_implementation == "PyPy") and extra == 'brotli'
cassandra-driver ; extra == 'cassandra'
python-consul ; extra == 'consul'
pydocumentdb (==2.3.2) ; extra == 'cosmosdbsql'
couchbase ; extra == 'couchbase'
couchbase-cffi ; (platform_python_implementation == "PyPy") and extra == 'couchbase'
pycouchdb ; extra == 'couchdb'
Django (>=1.8) ; extra == 'django'
boto3 (>=1.4.6) ; extra == 'dynamodb'
elasticsearch ; extra == 'elasticsearch'
eventlet (>=0.24.1) ; extra == 'eventlet'
gevent ; extra == 'gevent'
librabbitmq (>=1.5.0) ; extra == 'librabbitmq'
backports.lzma ; (python_version < "3.3") and extra == 'lzma'
pylibmc ; extra == 'memcache'
pymongo[srv] (>=3.3.0) ; extra == 'mongodb'
msgpack ; extra == 'msgpack'
python-memcached ; extra == 'pymemcache'
pyro4 ; extra == 'pyro'
redis (>=3.2.0) ; extra == 'redis'
riak (>=2.0) ; extra == 'riak'
boto3 (>=1.4.6) ; extra == 's3'
softlayer-messaging (>=1.0.3) ; extra == 'slmq'
ephem ; extra == 'solar'
sqlalchemy ; extra == 'sqlalchemy'
boto3 (>=1.4.6) ; extra == 'sqs'
pycurl ; extra == 'sqs'
tblib (>=1.3.0) ; extra == 'tblib'
PyYAML (>=3.10) ; extra == 'yaml'
kazoo (>=1.3.1) ; extra == 'zookeeper'
zstandard ; extra == 'zstd'
```

```
$ pip freeze --all
amqp==2.5.0
atomicwrites==1.3.0
attrs==19.1.0
billiard==3.6.0.0
celery==4.3.0
dumb-init==1.2.2
importlib-metadata==0.17
kombu==4.6.0
more-itertools==7.0.0
packaging==19.0
pip==19.1.1
pluggy==0.12.0
py==1.8.0
pyparsing==2.4.0
pytest==4.6.2
pytz==2019.1
setuptools==41.0.1
six==1.12.0
vine==1.3.0
wcwidth==0.1.7
wheel==0.33.4
zipp==0.5.1
```

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 1, 2019, 15:45

Yikes! It looks like importlib_metadata is only ever returning the first value from Requires-Dist. Definitely a problem. The project will probably need an expanded test case to capture this missed expectation. Is this something you would like to work on, @asottile ?

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 1, 2019, 15:45

assigned to @asottile

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 1, 2019, 16:23

assigned to @jaraco and unassigned @asottile

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 1, 2019, 16:23

I took a quick look and I think a fix should be fairly straightforward. The fixtures already expose multiple requirements; it's just that the tests don't fully exercise that expectation.

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 1, 2019, 16:24

mentioned in commit 79c6670c106fb7e75f7a98e3358f6ee3bab2a268

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 1, 2019, 16:30

mentioned in commit adf34db3cd5ee47ecbec403a90d5594d52db002f

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 1, 2019, 16:30

mentioned in merge request !86

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 2, 2019, 12:12

closed via commit 190789b7be5c2ece9c2a5d5cc6cf0504c2b6dc91

**Comment by jaraco** [comment]
In GitLab by [@jaraco](https://gitlab.com/jaraco) on Sep 2, 2019, 12:13

mentioned in commit 190789b7be5c2ece9c2a5d5cc6cf0504c2b6dc91
